netbox/utilities/mixins.py

```diff
@@ -0,0 +1,54 @@
+from django.db.models.query_utils import DeferredAttribute
+
+
+class Tracker(object):
+    def __init__(self, instance):
+        self.instance = instance
+        self.newly_created = False
+        self.changed = {}
+        self.tracked_fields = self.instance.change_tracking_fields
+
+
+class TrackingModelMixin(object):
+    change_tracking_fields = []
```

Comment:
Because this is declared as a class attribute, the list is being shared across _all_ models:

```python
>>> ConsolePort.change_tracking_fields
['device_id', 'virtual_machine_id']
```

Instead of defining this on the model, could we create a registry entry for mapping models to their tracked fields? This would probably be a bit cleaner.

Reply:
made an instance var